pkg/kubelet/kubelet_node_status_test.go

```diff
@@ -1526,6 +1526,80 @@ func TestUpdateDefaultLabels(t *testing.T) {
 	}
 }
 
+func TestReconcileExtendedResource(t *testing.T) {
+	testKubelet := newTestKubelet(t, false /* controllerAttachDetachEnabled */)
+	testKubelet.kubelet.kubeClient = nil // ensure only the heartbeat client is used
+	extendedResourceName1 := v1.ResourceName("test.com/resource1")
+	extendedResourceName2 := v1.ResourceName("test.com/resource2")
+
+	cases := []struct {
+		name         string
+		existingNode *v1.Node
+		expectedNode *v1.Node
+		needsUpdate  bool
+	}{
+		{
+			name: "no update needed without extended resource",
+			existingNode: &v1.Node{
+				Status: v1.NodeStatus{
```

Comment:
Should this test check Allocatable?

Reply:
Ah my bad. I did extend the test to check allocatable but forgot to push the change.

Reply:
Ah you just updated it!